modules/ballerina-http/src/main/java/org/ballerinalang/net/http/session/HTTPSession.java

```diff
@@ -58,22 +59,26 @@ public String getId() {
 
     @Override
     public void setAttribute(String attributeKey, BValue attributeValue) {
-        checkValidity();
         attributeMap.put(attributeKey, attributeValue);
     }
 
     @Override
     public BValue getAttributeValue(String attributeKey) {
-        checkValidity();
         return attributeMap.get(attributeKey);
     }
 
     @Override
     public String[] getAttributeNames() {
-        checkValidity();
         return attributeMap.keySet().toArray(new String[attributeMap.size()]);
     }
 
+    @Override
+    public BMap<String, BValue> getAttributes() {
+        if (!attributeMap.isEmpty()) {
+            return attributeMap;
+        }
+        return null;
```

Comment:
IMO we should always return the map, regardless its empty or not.

Reply:
Ack